plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/organize/imports/N4JSReferencesFilter.java

```diff
@@ -18,6 +25,29 @@
 
 	@Override
 	public boolean test(ReferenceProxyInfo proxyInfo) {
-		return true;
+		ResourceType resourceType = ResourceType.getResourceType(proxyInfo.eobject);
+		switch (resourceType) {
+		case N4JSX:
+			return !n4jxIsToBeIgnored(proxyInfo);
+
+		default:
+			return true;
+		}
```

Comment:
Interface `IReferenceFilter` and all implementing classes were only a work-around required because N4JSX was a separate Xtext language. So I propose to remove that interface, all implementing classes, and all related code (esp. the infrastructure for registering implementing classes). The above special handling should then be added to where the IReferenceFilter is used in ordinary N4JS code.

(this is very similar to `UnsupportedExpressionTypeHelper`, above)

Reply:
removed